In [17]:
from scipy import polyval, polyfit, sqrt
import pandas as pd
import numpy as np

In [2]:
from bokeh.plotting import figure, show
from bokeh.io import output_notebook
output_notebook()

Loading BokehJS ...

# Homework 7



## Power Estimation

In this homework, we're attempting to more accurately predict the power of a spacecraft using a regression analysis of previous space craft. The following spacecraft were used for this analysis:

* Mars Reconnasaince Orbiter (MRO)
* Lunar Reconnassaince Orbiter (LRO)
* Mars Odyssey 2001 (MO)
* Cassini Spacecraft (CS)
* New Horizons (NH)

*Note: JUNO was also supposed to be a part of this analysis, however, I was unable to determine the payload power for this spacecraft*

<table>
    <tr><th>Spacecraft</th><th>Payload Power (W)</th><th>Total Power (W)</th></tr>
    <tr><td>MRO</td>       <td>65</td>               <td>2000</td>           </tr>
    <tr><td>LRO</td>       <td>112</td>              <td>685</td>            </tr>
    <tr><td>MO</td>        <td>73</td>               <td>750</td>            </tr>
    <tr><td>CS</td>        <td>348</td>              <td>885</td>            </tr>
    <tr><td>NH</td>        <td>12</td>               <td>180</td>            </tr>
</table>

In [10]:
payload_power = [189, 112, 73, 348, 12]
total_power = [600, 685, 750, 885, 180]

# Linear Regression:
(ar, br) = polyfit(payload_power, total_power, 1)
r = polyval([ar, br], payload_power)

In [12]:
p = figure(plot_width=600, plot_height=600, title="Payload vs. Total Power (in W)")
p.scatter(payload_power, total_power, size=3, color="#3A5785", alpha=0.6)
p.line(payload_power, r)
show(p)

In [8]:
print("Linear Regression equation = {:.4}x + {:.4}".format(ar, br))

Linear Regression equation = 1.489x + 401.3


Using this linear regression we can estimate the total power based on the linear regression model. 

In [13]:
def sc_total_power(pl_power):
    return 1.489 * pl_power + 401.3

Current estimates has the payload power at ~72.1 W Peak Power.  Given this, we can estimate the total power required:

In [15]:
t_power = sc_total_power(72.1)
t_power

508.6569

## Power Design


1. Power Requirements
2. Solar Arrays
3. Batteries
4. Mass and Power Estimates for subsystems
5. Power system block diagram

### Power Requirements

Given that the payload power is 72.1 W, the estimated total power is 508.6569 W.  From this, we allocate 436.55 W to the subsystem power.  

### Power Estimates for subsystems

Using the subsystem % estimates from the lectures, we get the following:

<table>
<tr><th>Subsystem</th> <th>% of Subsytem Total</th></tr>
<tr><td>Thermal</td>   <td>28</td></tr>
<tr><td>Attitude</td>  <td>20</td></tr>
<tr><td>Power</td>     <td>10</td></tr>
<tr><td>C and DH</td>  <td>17</td></tr>
<tr><td>Comm</td>      <td>23</td></tr>
<tr><td>Propulsion</td><td>1 </td></tr>
<tr><td>Mech</td>      <td>1 </td></tr>
</table>

From the total power estimates, we can now estimate the power for each subsystem:

In [20]:
subsystem_power = t_power - 72.1
thermal_power = subsystem_power * 0.28
attitude_power = subsystem_power * 0.20
power_power = subsystem_power * 0.10
cdh_power = subsystem_power * 0.17
comm_power = subsystem_power * 0.23
prop_power = subsystem_power * 0.01
mech_power = subsystem_power * 0.01

index = ['Thermal', 'Attitude', 'Power', 'C&DH', 'Comm', 'Propulsion', 'Mech']
data = [thermal_power, attitude_power, power_power, cdh_power, comm_power, prop_power, mech_power]

pd.DataFrame(data, index=index, columns=['Estimated Power (W)'])

Estimated Power (W)
Thermal              122.235932
Attitude              87.311380
Power                 43.655690
C&DH                  74.214673
Comm                 100.408087
Propulsion             4.365569
Mech                   4.365569